docs/components/codeblock.md

```diff
@@ -0,0 +1,357 @@
+
+:::danger Experimental Component
+`KCodeBlock` is an experimental component. The component’s design and overall look and feel may change until it is in stable status.
+:::
+
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-default"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+```
+
+## Props
+
+### id
+
+* **Type**: `string`
+* **Required**: yes
+
+An ID used to identify form elements like the search input and its label.
+
+**Note**: If you use multiple instances of `KCodeBlock` in a document, you need to provide unique ID values.
+
+### code
+
+* **Type**: `string`
+* **Required**: yes
+
+The code that will be rendered as a text node inside of a `<pre><code></code></pre>` fragment.
+
+### language
+
+* **Type**: `string`
+* **Required**: yes
+
+The syntax language of `props.code` (e.g. `'json'`).
+
+### isSearchable
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Shows an actions bar with a search input and related action buttons.
+
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-searchable"
+  :code="code"
+  language="json"
+  is-searchable
+/>
+```
+
+### isProcessing
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Allows controlling the processing state from outside the component. This allows a parent component to show the processing icon when it’s, for example, currently syntax highlighting the code.
+
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-searchable
+  is-processing
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-is-processing"
+  :code="code"
+  language="json"
+  is-processing
+/>
+```
+
+### query
+
+* **Type**: `string`
+* **Required**: no
+* **Default**: `''`
+
+Used as the initial value of the search input. Can be used to initialize a code block with a query which was read from client storage.
+
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="no"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-query"
+  :code="code"
+  language="json"
+  is-searchable
+  query="no"
+/>
+```
+
+### showCopyButton
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `true`
+
+Controls whether to show a copy button which copies the original code (i.e. the value of `props.code`) to the clipboard.
+
+<KCodeBlock
+  id="code-block-show-copy-button"
+  :code="code"
+  language="json"
+  :show-copy-button="false"
+/>
+
+```vue
+<KCodeBlock
+  id="code-block-show-copy-button"
+  :code="code"
+  language="json"
+  :show-copy-button="false"
+/>
+```
+
+### showLineNumberLinks
+
+* **Type**: `boolean`
+* **Required**: no
+* **Default**: `false`
+
+Controls whether to add links to the current line number.
```

Comment:
This one confuses me a bit. If you're on a URL that already has a hash (e.g. the VuePress docs site) these links aren't functional. Ideas?

Reply:
This is ironically why I introduced the prop in the first place: Currently (and I plan to change this), Kuma GUI uses vue router’s hash history mode meaning that virtually all URLs have a fragment identifier in the URL so there’s no way to use anchors like this.

On the other hand, for sites where this isn’t the case, you can get similar behavior to what GitHub does in code views (e.g. link to specific lines). This is naturally in conflict with pages that have many anchors for other use cases like documentation pages, but I think that’s fine: it’s not a severe conflict. Anchor conflicts are easily avoided using a strategic value for `props.id` and these anchors don’t prevent other anchors from working.

Is there a specific concern you want to address?

Reply:
Matches what I was saying.

Can we add a note under this prop that outlines a use-case when it needs to be disabled, e.g. if the page utilizes hash navigation, etc.?

Reply:
Yes, I’ve added a sentence for this now.